orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/DetermineHealthProvidersTask.java

```diff
@@ -0,0 +1,99 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks;
+
+import com.google.common.collect.ImmutableMap;
+import com.netflix.frigga.Names;
+import com.netflix.spinnaker.orca.DefaultTaskResult;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.RetryableTask;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.clouddriver.utils.CloudProviderAware;
+import com.netflix.spinnaker.orca.front50.Front50Service;
+import com.netflix.spinnaker.orca.front50.model.Application;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+@Component
+public class DetermineHealthProvidersTask implements RetryableTask, CloudProviderAware {
+  private static final Logger log = LoggerFactory.getLogger(DetermineHealthProvidersTask.class);
+  private static final Map<String, String> HEALTH_PROVIDER_NAMES_BY_PLATFORM = ImmutableMap
+    .<String, String>builder()
+    .put("aws", "Amazon")
+    .put("gce", "Google")
+    .put("openstack", "Openstack")
+    .put("titus", "Titus")
+    .build();
+
+  @Autowired
+  Front50Service front50Service;
+
+  @Override
+  public TaskResult execute(Stage stage) {
+    if (stage.getContext().containsKey("interestingHealthProviderNames")) {
+      // should not override any stage-specified health providers
+      return new DefaultTaskResult(ExecutionStatus.SUCCEEDED);
+    }
+
+    String platformSpecificHealthProviderName = HEALTH_PROVIDER_NAMES_BY_PLATFORM.get(getCloudProvider(stage));
+    if (platformSpecificHealthProviderName == null) {
+      log.warn("Unable to determine platform health provider for unknown cloud provider '{}'", getCloudProvider(stage));
+      return new DefaultTaskResult(ExecutionStatus.SUCCEEDED);
+    }
+
+    try {
+      String applicationName = (String) stage.getContext().get("application");
+      if (applicationName == null && stage.getContext().containsKey("serverGroupName")) {
+        applicationName = Names.parseName((String) stage.getContext().get("serverGroupName")).getApp();
+      } else if (applicationName == null && stage.getContext().containsKey("asgName")) {
+        applicationName = Names.parseName((String) stage.getContext().get("asgName")).getApp();
+      } else if (applicationName == null && stage.getContext().containsKey("cluster")) {
+        applicationName = Names.parseName((String) stage.getContext().get("cluster")).getApp();
+      }
+
+      Application application = front50Service.get("default", applicationName);
+      if (application.platformHealthOnly == Boolean.TRUE && application.platformHealthOnlyShowOverride != Boolean.TRUE) {
```

Comment:
Sure ... I think it gets tricky with:

`platformHelathOnly: true` and `platformHealthOnlyShowOverride: true` and then the user actually deselects the "Use Platform Health" checkbox when the go to run the operation.

In such a scenario, `interestingHealthProviderNames` would not come across in the request.

This is quite an edge-case ... one that I'm not too sure is all that relevant. Let me do some quick analysis on all our applications to see how many have `platformHealth: true` and `platformHealthOnlyShowOverride: true`. I still think it's confusing that we allow both to be selected.

Are you thinking we should update deck and prevent both options from being selected? I could get behind that.